modules/gapi/src/executor/gexecutor.cpp

```diff
@@ -421,7 +422,10 @@ void cv::gimpl::GExecutor::reshape(const GMetaArgs& inMetas, const GCompileArgs&
     ade::passes::PassContext ctx{g};
     passes::initMeta(ctx, inMetas);
     passes::inferMeta(ctx, true);
-    m_ops[0].isl_exec->reshape(g, args);
+    for (auto& op : m_ops)
+    {
+        op.isl_exec->reshape(g, args);
+    }
```

Comment:
Note this may be not enough.

`GExecutor` owns its own resource magazine and when metadata changes, objects may need to be reallocated. Check the `initResource()` which is currently called in the constructor only.

There may be other tricky parts w.r.t. resource management, too.

Reply:
Done